docs/Configuration.md

```diff
@@ -44,27 +44,178 @@ Below is a conceptual example of what happens during normalization.
+        "typeMatchExpression": {
+            "value" : "$..[?(@heartRate)]",
+            "language": "JsonPath"
+        },
+        ...
+    }
+```
+
+If it's desired to simply use the default expression language, the expression alone may be supplied:
+
+```
+"templateType": "CalculatedContentTemplate",
+    "template": {
+        "typeName": "heartrate",
+        "typeMatchExpression": "$..[?(@heartRate)]",
+        ...
+    }
+```
+
+It is also possible to explicitly set the default expression language that will be used:
+
+```
+"templateType": "CalculatedContentTemplate",
+    "template": {
+        "typeName": "heartrate",
+        "defaultExpressionLanguage": "JsonPath",
+        "typeMatchExpression": "$..[?(@heartRate)]",
+        ...
+    }
+```
+### Expression Languages
+
+When specifying the language to use for the expression, the below values are valid:
+| Expression Language | Value        |
+|---------------------|--------------|
+| JSON Path           | **JsonPath** |
+| JmesPath            | **JmesPath** |
+
+More information on JSON Path can be found [here](https://goessner.net/articles/JsonPath/). The [CalculatedContentTemplete](#CalculatedContentTemplate) uses the [JSON .NET implementation](https://www.newtonsoft.com/json/help/html/QueryJsonSelectTokenJsonPath.htm) for resolving JSON Path expressions. Additional examples can be found in the [unit tests](../test/Microsoft.Health.Fhir.Ingest.UnitTests/Template/JsonPathContentTemplateTests.cs).
+
+More information on JmesPath can be found [here](https://jmespath.org/specification.html). [CalculatedContentTemplete](#CalculatedContentTemplate) uses the [JmesPath .NET implementation](https://github.com/jdevillard/JmesPath.Net) for resolving JmesPath expressions. In addtion to the functions provided as part of the specification a set of custom functions are also available for use. More information on them can be found [here](./CustomFunctions.md). Source code for the functions can be found TODO. 
+
+### Matched Token
+The **TypeMatchExpression** is evaluated against the incoming EventData payload. If a matching JToken is found the template is considered a match. All subsequent expressions are evaluated against a new JToken which contains both the original EventData payload as well as the extracted JToken matched here. The extracted JToken will be available as the property __matchedToken__.
+
+Given the following:
+
+_Message_
+
+```
+{
+  "Properties": {
+    "deviceId": "device123"
+  },
+  "SystemProperties": {},
+  "Body": [
+    {
+      "systolic": "120",
+      "diastolic": "80",
+      "date": "2021-07-13T17:29:01.061144Z"
+    },
+    {
+      "systolic": "122",
+      "diastolic": "82",
+      "date": "2021-07-13T17:28:01.061122Z"
+    }
+  ]
+}
+```
+
+_Template_
+
+```
+"templateType": "CalculatedContentTemplate",
+    "template": {
+        "typeName": "heartrate",
+        "typeMatchExpression": "$..[?(@systolic && @diastolic)]",
+        ...

```

Comment:
Recommend having a whole JSON content. as most of the customers like to copy and paste it while testing.